modules/net-vpc/subnets.tf

```diff
@@ -79,6 +79,10 @@ locals {
     { for s in var.subnets_psc : "${s.region}/${s.name}" => s },
     { for k, v in local._factory_subnets : k => v if v.purpose == "PRIVATE_SERVICE_CONNECT" }
   )
+  subnets_global_proxy_only = merge(
+    { for s in var.subnets_psc : "${s.region}/${s.name}" => s },
```

Comment:
I guess you wanted to introduce new variable here, instead reuse `var.subnets_psc`?

Reply:
Also, couldn't we re-use `var.subnets_proxy_only` and have the filtering predicate have something like `contains(v.purpose, ["REGIONAL_MANAGED_PROXY", "GLOBAL_MANAGED_PROXY"])`?